pcsx2-qt/Debugger/SymbolTree/SymbolTreeWidgets.cpp

```diff
@@ -0,0 +1,1021 @@
+	const SymbolWork& child_work,
+	SymbolTreeNode*& prev_group,
+	const SymbolWork*& prev_work,
+	const SymbolFilters& filters)
+{
+	bool group_exists =
+		prev_group &&
+		child_work.section == prev_work->section &&
+		(!filters.group_by_module || child_work.module_symbol == prev_work->module_symbol);
+	if (group_exists)
+	{
+		prev_group->emplaceChild(std::move(child));
+		return nullptr;
+	}
+
+	std::unique_ptr<SymbolTreeNode> group_node = std::make_unique<SymbolTreeNode>();
+	if (child_work.section)
+	{
+		group_node->tag = SymbolTreeNode::GROUP;
+		group_node->name = QString::fromStdString(child_work.section->name());
+		if (child_work.section->address().valid())
+			group_node->location = SymbolTreeLocation(SymbolTreeLocation::MEMORY, child_work.section->address().value);
+	}
+	else
+	{
+		group_node->tag = SymbolTreeNode::UNKNOWN_GROUP;
+		group_node->name = tr("(unknown section)");
+	}
+
+	group_node->emplaceChild(std::move(child));
+	child = std::move(group_node);
+
+	prev_group = child.get();
+	prev_work = &child_work;
+
+	return child;
+}
+
+std::unique_ptr<SymbolTreeNode> SymbolTreeWidget::groupByModule(
+	std::unique_ptr<SymbolTreeNode> child,
+	const SymbolWork& child_work,
+	SymbolTreeNode*& prev_group,
+	const SymbolWork*& prev_work,
+	const SymbolFilters& filters)
+{
+	bool group_exists =
+		prev_group &&
+		child_work.module_symbol == prev_work->module_symbol;
+	if (group_exists)
+	{
+		prev_group->emplaceChild(std::move(child));
+		return nullptr;
+	}
+
+	std::unique_ptr<SymbolTreeNode> group_node = std::make_unique<SymbolTreeNode>();
+	if (child_work.module_symbol)
+	{
+
+		group_node->tag = SymbolTreeNode::GROUP;
+		group_node->name = QString::fromStdString(child_work.module_symbol->name());
+		if (child_work.module_symbol->address().valid())
+			group_node->location = SymbolTreeLocation(SymbolTreeLocation::MEMORY, child_work.module_symbol->address().value);
+		if (child_work.module_symbol->is_irx)
+		{
+			s32 major = child_work.module_symbol->version_major;
+			s32 minor = child_work.module_symbol->version_minor;
+			group_node->name += QString(" v%1.%2").arg(major).arg(minor);
+		}
+	}
+	else
+	{
+		group_node->tag = SymbolTreeNode::UNKNOWN_GROUP;
+		group_node->name = tr("(unknown module)");
+	}
+
+	group_node->emplaceChild(std::move(child));
+	child = std::move(group_node);
+
+	prev_group = child.get();
+	prev_work = &child_work;
+
+	return child;
+}
+
+void SymbolTreeWidget::setupMenu()
+{
+	m_context_menu = new QMenu(this);
+
+	QAction* copy_name = new QAction(tr("Copy Name"), this);
+	connect(copy_name, &QAction::triggered, this, &SymbolTreeWidget::onCopyName);
+	m_context_menu->addAction(copy_name);
+
+	QAction* copy_location = new QAction(tr("Copy Location"), this);
+	connect(copy_location, &QAction::triggered, this, &SymbolTreeWidget::onCopyLocation);
+	m_context_menu->addAction(copy_location);
+
+	m_context_menu->addSeparator();
+
+	m_rename_symbol = new QAction(tr("Rename Symbol"), this);
```

Comment:
It might be better UX to check if the user has a selection and conditionally add context menu options if they do. Currently it gives the option to rename/copy/goto even if no items exist.  [Here's an example](https://github.com/PCSX2/pcsx2/blob/master/pcsx2-qt/Debugger/CpuWidget.cpp#L325) where we do this.
 
By no means a big issue, just wanted to mention this.

Reply:
I prefer disabling the menu items instead of removing them entirely so you can get a full picture of all the options that are available. That said, I don't do this for the goto in memory/goto in disassembly options currently so that's something I could add.

Reply:
I'm now disabling the goto in disassembly/memory view options when it doesn't make sense to use them.